### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by holiber**

### CODE REVIEW

File: app/services/platforms/twitch.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export class TwitchService
   };
 
   // Streamlabs Production Twitch OAuth Client ID
-  clientId = '8bmp6j83z5w4mepq0dn0q1a7g186azi';
+  clientId = '3eoucd9qwxqh7pu3l0e3rttomgrov2';
```
Comment (trevorrl) [inline app/services/platforms/twitch.ts:109]: assuming this is the beta client ID so will want to change this back as well

File: app/components-react/shared/PlatformLogo.tsx [hunk 2]
```diff
@@ -3,22 +3,65 @@ import { TPlatform } from '../../services/platforms';
...
+    alignItems: 'center',
+    justifyContent: 'center',
+  };
+
+  return (
+    <div style={wrapperStyles}>
+      <i
+        className="fab tiktok"
+        style={{ width: `${size / 2}px` }}
+        dangerouslySetInnerHTML={{ __html: getTikTokSvg(color) }}
```
Comment (gettinToasty) [inline app/components-react/shared/PlatformLogo.tsx:63]: why are we using `i` with `dangerouslySetInnerHTML` here? you can just use an SVG like any other React or Vue component, see `KevinSVG.tsx` for an example

Comment (holiber) [reply app/components-react/shared/PlatformLogo.tsx:63]: We don't ask to React to parse SVG nodes and track dom changes if we provide SVG as simple text. There are no benefits in this particular example but you might find performance issues for a bigger SVG markup,

Comment (holiber) [reply app/components-react/shared/PlatformLogo.tsx:63]: Solution B always forks faster than solution A, and solution A has no advantages against solution B. I don't have any strong reasons to use solution A in this case.

Comment (holiber) [reply app/components-react/shared/PlatformLogo.tsx:63]: >  but if there is no benefit...
There is a benefit but it's not noticeable. You will not feel the difference between `1ms` rendering and `10ms` rendering for one SVG. But technically it's `x10` difference. And when you're rendering SVG with React it really works as much slower than static rendering.

File: app/services/platforms/tiktok.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  readonly displayName = 'TikTok';
+  readonly capabilities = new Set<TPlatformCapability>([]);
+
+  // support only donation widgets for now
+  readonly widgetsWhitelist = [
+    WidgetType.AlertBox,
+    WidgetType.DonationGoal,
+    WidgetType.DonationTicker,
+    WidgetType.TipJar,
+  ];
```
Comment (gettinToasty) [inline app/services/platforms/tiktok.ts:30]: you can use the `platforms` field of `WidgetDisplayData` in `widgets-data` to accomplish this :)

File: app/components/windows/go-live/CommonPlatformFields.tsx [hunk 4]
```diff
@@ -176,7 +176,7 @@ export default class CommonPlatformFields extends TsxComponent<Props> {
             <div>
               {/*TITLE*/}
               <HFormGroup
-                value={fields.title}
+                value={fields['title']}
```
Comment (gettinToasty) [inline app/components/windows/go-live/CommonPlatformFields.tsx:179]: not blocking but wondering why this change was made

File: app/services/platforms/tiktok.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * prepopulate channel info and save it to the store
+   */
+  async prepopulateInfo(): Promise<void> {
+    this.SET_PREPOPULATED(true);
+  }
+
+  async fetchViewerCount(): Promise<number> {
+    // no API
+    return 0;
```
Comment (gettinToasty) [inline app/services/platforms/tiktok.ts:68]: wonder if we just want to hide view count for tik tok here if there isn't an api for it

File: app/services/platforms/tiktok.ts (hunk 3 above)
Comment (holiber) [reply app/services/platforms/tiktok.ts:30]: I found the `platforms` field for events there but don't see any example for widgets

File: app/components/windows/go-live/CommonPlatformFields.tsx (hunk 4 above)
Comment (holiber) [reply app/components/windows/go-live/CommonPlatformFields.tsx:179]: There is a TS type issue that I'm not sure how to resolve with massive changes in type description. `fields.title` used to be a field for all platforms but it's not true now. However, we never show `CommonPlatformFields.tsx` for TikTok so should be no real errors here

File: app/services/platforms/tiktok.ts (hunk 5 above)
Comment (holiber) [reply app/services/platforms/tiktok.ts:70]: We still show a `ViewerCount` for other platforms in multiplatform mode. But I agree it's forth to handle it for TikTok separately. I created an Asana tiket

File: app/services/platforms/tiktok.ts (hunk 3 above)
Comment (gettinToasty) [reply app/services/platforms/tiktok.ts:30]: it currently only exists for platform-specific widgets, but can be extended to include all platforms except tiktok where needed
```
  [WidgetType.SupporterGoal]: {
    name: $t('Supporter Goal'),
    description: $t('Set a goal for your viewers to help you reach.'),
    demoVideo: false,
    demoFilename: 'source-follower-goal.png',
    supportList: [$t('Facebook Supporters')],
    platforms: new Set(['facebook']),
  },
```

**Review by andycreeth — APPROVED** [review]
